.github/workflows/ci.yml

```diff
@@ -12,6 +12,7 @@ env:
   NXF_ANSI_LOG: false
   NXF_SINGULARITY_CACHEDIR: ${{ github.workspace }}/.singularity
   NXF_SINGULARITY_LIBRARYDIR: ${{ github.workspace }}/.singularity
+  NCBI_API_KEY: c47a089dc53f883cca01f105897a88bab108
```

Comment:
Can we store this key as a github secret variable? 
https://github.com/sanger-tol/genomeassembly/settings/secrets/actions

I don't think NCBI query can pick up this variable in this way. Can you check how this key can be used for the query?

Reply:
I tried to implement it as described here
https://github.com/NCBI-Hackathons/EDirectCookbook/issues/42

When the variable is set locally the pipeline runs a few times and the error didn't appear (although the error rarely came up locally). Maybe if setting it through the secrets will work but the page is not available for me and I don't have the 'Secrets and variables' section for this repo as in the guideline  https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions
I don't have admin rights for this repo:

    "permissions": {
      "admin": false,
      "maintain": true,
      "push": true,
      "triage": true,
      "pull": true
    },
    "role_name": "maintain"

@muffato could you perhaps elevate my rights for this repo?

Reply:
Set the API key as a secret won't help anything , just try to hide it from the public.

Without this key in your local environment, probably it won't exceed NCBI's query limit but probably lots of request in Github as a whole.

In this way, you only create an environment variable in the host system, but your NCBI query inside the container. I think you need to pass this variable inside the container.

Reply:
Here is how we do things in the genome-note pipeline, in the module that writes data back to the genome-note database using the API key provided by ToLP:

- The variable is declared as a Nextflow secret in the module: https://github.com/sanger-tol/genomenote/blob/public_dev/modules/local/write_to_database.nf#L3
- It is used like a normal environment variable by the script. Either implicitly (it will be picked up by the command/tool without you having to pass it on the command-line), or explicitly like in https://github.com/sanger-tol/genomenote/blob/public_dev/modules/local/write_to_database.nf#L25

Nextflow takes care of passing the secret to the container.
The key *has* to be defined as a Nextflow secret before starting the pipeline. We've got these instructions in `docs/usage.md`: https://github.com/sanger-tol/genomenote/blob/public_dev/docs/usage.md?plain=1#L32-L36
However, I can't see how it's being used in the CI. I remember discussing with Beth that the secret could be defined in Nextflow Tower to use in the "full test" run on the farm. I'll ask her tomorrow.

That said, here is how it should be set up for the regular test that runs on GitHub.

1. I've added the key as a GitHub Action secret for this reposiory.
2. Add this into`.github/workflows/ci.yml` before the "Run pipeline with test data" block
    ```yml
      - name: Set up nextflow secrets
        nextflow secrets set NCBI_API_KEY ${{ secrets.NCBI_API_KEY }}
    ```